### ISSUE TITLE
MD-04: Chạy thử huấn luyện 1 epoch để bắt lỗi runtime

### ISSUE THREAD

**Issue by GiaBaoK46DLU** [GiaBaoK46DLU/AgriXAI#13]
## Mô tả

Code trong `model/` **đã commit nhưng chưa từng chạy** — cần `torch` và bộ PlantVillage nên tới giờ mới thử được.

Chạy **đúng 1 epoch** để lộ hết lỗi runtime trước khi bỏ ra nhiều giờ cho lần huấn luyện thật. Ở bước này **chưa quan tâm accuracy**.

```bash
python -m src.training.train --config configs/default.yaml   # tạm sửa epochs: 1
```

## Tiêu chí hoàn thành

- [ ] Hết một epoch **không lỗi**
- [ ] Có file checkpoint ghi ra trong `checkpoints/`
- [ ] Checkpoint tự mô tả đủ: `arch`, `class_keys`, `image_size`, `dropout`, `state_dict`, `model_version`, `epoch`, `val_acc` — thiếu trường nào thì `load_checkpoint` sau này sẽ hỏng
- [ ] Thứ tự `class_keys` trong checkpoint **khớp đúng thứ tự mảng `diseases`** trong `shared/data/tomato_diseases.json`. Sai chỗ này thì mọi nhãn trả về đều lệch mà **không báo lỗi gì**
- [ ] Có file `outputs/logs/history-*.json`
- [ ] Ghi lại **thời gian thật của 1 epoch** vào issue, nhân với 25 để ước tính lần chạy đầy đủ
- [ ] Nếu có GPU: xác nhận nó đang được dùng thật (`nvidia-smi` thấy tiến trình python)
- [ ] Trả `epochs` về `25` sau khi thử xong

## Gợi ý triển khai

Lỗi hay gặp ở lần chạy đầu: `num_workers: 4` gây treo trên Windows nếu code không bọc trong `if __name__ == "__main__"`; hết VRAM thì giảm `batch_size` xuống 16.

## Phụ thuộc

- **Chặn bởi #12**
- Cần môi trường từ #10

## Nhánh

`bao/task-13-model-smoke-train` (nếu phải sửa lỗi)
